save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/topbar/TopBar.kt

```diff
@@ -0,0 +1,104 @@
+/**
+ * Top bar of web page
+ */
+
+@file:Suppress("FILE_NAME_MATCH_CLASS", "FILE_WILDCARD_IMPORTS", "WildcardImport")
+
+package com.saveourtool.save.frontend.components.topbar
+
+import com.saveourtool.save.frontend.externals.fontawesome.*
+import com.saveourtool.save.info.UserInfo
+
+import csstype.ClassName
+import csstype.Width
+import dom.html.HTMLButtonElement
+import history.Location
+import react.*
+import react.dom.html.ButtonHTMLAttributes
+import react.dom.html.ButtonType
+import react.dom.html.ReactHTML.button
+import react.dom.html.ReactHTML.nav
+import react.router.useLocation
+
+/**
+ * [Props] of the top bor component
+ */
+external interface TopBarProps : PropsWithChildren {
+    /**
+     * Currently logged in user or null
+     */
+    var userInfo: UserInfo?
+
+    /**
+     * true if the device is mobile (screen is less 1000px)
+     */
+    var isMobileScreen: Boolean?
+}
+
+/**
+ * [Props] of the top bor component with [location]
+ */
+external interface TopBarPropsWithLocation : TopBarProps {
```

Comment:
Why do you have`TopBarPropsWithLocation` in this file but do not use it here?

Reply:
Moreover, is there any case when you use both `location` and `userInfo` or something else from `TopBarProps`? If no, please, do not inherit the `TopBarProps` interface. To my mind it might be even better to create three (or four idk) interfaces for each component in case we might want to pass something specific to the child component